waffle_hub/dataset/dataset.py

```diff
@@ -603,7 +594,7 @@ def merge(
                 io.remove_directory(merged_ds.dataset_dir)
             raise e
 
-        return Dataset.load(name, root_dir)
+        return merged_ds
```

Comment:
What's the difference?

Reply:
> What's the difference?

The previous cached_property value did not behave as intended, resulting in two different outcomes.
Now that it has been fixed, I changed it because it would be better to eliminate the time it takes to reload

Reply:
Do we need the test code for `cached_property`?

I knew that load can bring some extra time, but I though it is more safe.

Reply:
> Do we need the test code for `cached_property`?
> 
> I knew that `load` can bring some extra time, but I though it is more safe.

I believe it is necessary to add cached_property as it is being used extensively. Please go ahead and add it.